ForumProject/projects/models.py

```diff
@@ -2,56 +2,181 @@
     PROJECT_STATUS_CHOICES = [
         ('open', 'Open'),
         ('closed', 'Closed'),
         ('pending', 'Pending'),
     ]
-    project_status = models.CharField(max_length=20, choices=PROJECT_STATUS_CHOICES, default='open')
+    status = models.CharField(max_length=20, choices=PROJECT_STATUS_CHOICES, default='open')
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+    duration = models.FloatField(blank=True, null=True, verbose_name='duration (months)')
+    budget_currency = models.CharField(max_length=3, blank=True, null=True)
+    budget_amount = models.IntegerField(blank=True, null=True)
+
+    class Meta:
+        verbose_name = 'Project'
+        verbose_name_plural = 'Projects'
+        ordering = ['startup', 'name']
+        constraints = [
+            models.UniqueConstraint(fields=['startup', 'name'],
+            name='unique_project_per_startup')
+        ]
+    
+    def save(self, *args, **kwargs):
+        """
+        Overriding save method to ensure a log is recorded when a Project is created or modified
+        """
+
+        # Determine if this is a new object or an existing one
+        is_new = self.pk is None
 
+
+        if not is_new:
+            action = 'Update of Project'
+            # Fetch the original instance from the database to compare with the current one
+            original = Project.objects.get(pk=self.pk)
+
+            # Compare attributes to identify changes
+            changes = []
+            for field in self._meta.fields:
+                field_name = field.name
+                if getattr(original, field_name) != getattr(self, field_name):
+                    changes.append(
+                        (field_name, getattr(original, field_name), getattr(self, field_name))
+                    )
+
+            if changes:
+                # Prepare the previous and modified states
+                previous_state = ', '.join([f'{field}: {old_value}' for field, old_value, _ in changes])
+                modified_state = ', '.join([f'{field}: {new_value}' for field, _, new_value in changes])
+        else:
+            action = 'Created Project'
+            previous_state = 'n/a'
+            modified_state = f'New Project, id: {self.pk}, name: {self.name}'
+
+        super().save(*args, **kwargs)
+        
+        # Create a new ProjectLog
+        current_date = datetime.now().date()
+        current_time = datetime.now().time()
+
+        ProjectLog.objects.create(
+            project=self,
+            project_title=self.name,
+            change_date=current_date,
+            change_time=current_time,
+            user_id=1,  # This is a placeholder
+            action=action,
+            previous_state=previous_state[:ProjectLog._meta.get_field('previous_state').max_length],
+            modified_state=modified_state[:ProjectLog._meta.get_field('modified_state').max_length]
+            )
```

Comment:
The method retrieves the original instance from the database for every update operation, which might be inefficient. Consider optimizing this process, possibly by caching frequently accessed data or limiting the database calls.
The method lacks error handling, especially when performing operations that might fail (like database queries)

Reply:
I gave it consideration and came to a conclusion that optimizing at this stage is hardly achievable. We can not know what Projects may be subject to frequent update operations, and if a Project gets updated, for proper logging purposes its original instance has to be retrieved, to create a log with "before-after" record. 
Errors handling is added. 
Please note the logic has been moved from `save()` method in models,py to views.py.